### PR TITLE
Feat/authentication/integrate vm into UI : integrate Supabase authentication into the authentication screens

### PR DESCRIPTION
**Pull request by charliemangano**
# Integrate Supabase authentication into the authentication screens
## Description
This PR integrates Supabase authentication into the authentication screens, making it possible (finally) for users to sign up and log in. It closes issue #49.

## Changes
This PR adds calls to the authentication view model when the user tries to log in or sign up, and to that end initialises a Supabase client in `MainActivity`.

#### Added
None
#### Modified
 - `app/build.gradle.kts`: enabled buildConfig and created variables for the Supabase URL and key in the `BuildConfig`
 - `app/src/main/java/com/android/periodpals/MainActivity.kt`: initialised and passed a Supabase client and authentication view model as argument to the rest of the app
 - `app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt`: added calls to the authentication view model
 - `app/src/main/java/com/android/periodpals/ui/authentication/SignUp.kt`: added calls to the authentication view model
#### Removed
 - `app/src/main/java/com/android/periodpals/model/SupabaseClient.kt`: replaced by initialisation in the main
## Dependencies Added
None
## Testing
See end-to-end tests (#48)

### CODE REVIEW

**Review by francelu — CHANGES_REQUESTED** [review]
## Summary
Great job integrating Supabase authentication into the sign-in and sign-up screens. Changes are needed.

## Important
### Code Quality
- Clean and readable; remove navigation logic intended for testing.

### Functionality
- Authentication flow works well; ensure navigation isn’t merged here.

## Steps before PR approval
* Remove unnecessary navigation logic.

**Review by agonzalez-r — APPROVED** [review]
## Summary
Very clean and organised code overall. Thank you for this implementation!
## Important
#### Code Quality: 
- Consistent code with clear naming conventions
- Well modularised with `AuthComponents.kt`
#### Functionality
- LGTM
#### Security 
- Not sure if its already done, but [this](https://supabase.com/docs/guides/api/securing-your-api) could be interesting 
## Testing
- LGTM 
## Steps before PR approved
LGTM

File: app/src/main/java/com/android/periodpals/ui/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -34,14 +34,15 @@ import androidx.compose.ui.unit.Dp
...
+import com.android.periodpals.ui.navigation.NavigationActions
 import com.android.periodpals.ui.navigation.TopAppBar
 import com.bumptech.glide.integration.compose.ExperimentalGlideComposeApi
 import com.bumptech.glide.integration.compose.GlideImage
 
 @OptIn(ExperimentalGlideComposeApi::class)
 // @Preview
 @Composable
-fun ProfileScreen() {
+fun ProfileScreen(navigationActions: NavigationActions) {
```
[the remark this answers is not in the document]
Comment (agonzalez-r) [reply app/src/main/java/com/android/periodpals/ui/profile/ProfileScreen.kt]: No, don't do that, it's still needed in order to call the function

**Review by Harrish92 — APPROVED** [review]
### Code Review

#### Summary
- The code is well-structured and the comments are clear and provide a good understanding of the code's functionality.

LGTM

**Review by francelu — APPROVED** [review]
LGTM.

**Pull request merged**